src/libraries/System.Formats.Tar/src/System/Formats/Tar/SubReadStream.cs

```diff
@@ -0,0 +1,180 @@
+            }
+        }
+
+        private void ThrowIfBeyondEndOfStream()
+        {
+            if (HasReachedEnd)
+            {
+                throw new EndOfStreamException();
+            }
+        }
+
+        public override int Read(byte[] buffer, int offset, int count) => Read(buffer.AsSpan(offset, count));
+
+        public override int Read(Span<byte> destination)
+        {
+            ThrowIfDisposed();
+            ThrowIfBeyondEndOfStream();
+
+            // parameter validation sent to _superStream.Read
+            int origCount = destination.Length;
+            int count = destination.Length;
+
+            if (_positionInSuperStream + count > _endInSuperStream)
+            {
+                count = (int)(_endInSuperStream - _positionInSuperStream);
+            }
+
+            Debug.Assert(count >= 0);
+            Debug.Assert(count <= origCount);
+
+            int ret = _superStream.Read(destination.Slice(0, count));
+
+            _positionInSuperStream += ret;
+            return ret;
+        }
+
+        public override int ReadByte()
+        {
+            byte b = default;
+            return Read(MemoryMarshal.CreateSpan(ref b, 1)) == 1 ? b : -1;
+        }
+
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            ValidateBufferArguments(buffer, offset, count);
+            return ReadAsync(new Memory<byte>(buffer, offset, count), cancellationToken).AsTask();
+        }
+
+        public override ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken = default)
+        {
+            ThrowIfDisposed();
+            ThrowIfBeyondEndOfStream();
+            return ReadAsyncCore(buffer, cancellationToken);
+        }
+
+        protected async ValueTask<int> ReadAsyncCore(Memory<byte> buffer, CancellationToken cancellationToken)
+        {
+            Debug.Assert(!_hasReachedEnd);
+
+            if (_positionInSuperStream > _endInSuperStream - buffer.Length)
+            {
+                buffer = buffer.Slice(0, (int)(_endInSuperStream - _positionInSuperStream));
+            }
+
+            int ret = await _superStream.ReadAsync(buffer, cancellationToken).ConfigureAwait(false);
+
+            _positionInSuperStream += ret;
+            return ret;
+        }
+
+        public override long Seek(long offset, SeekOrigin origin) => throw new InvalidOperationException(SR.IO_NotSupported_UnseekableStream);
+
+        public override void SetLength(long value) => throw new InvalidOperationException(SR.IO_NotSupported_UnseekableStream);
+
+        public override void Write(byte[] buffer, int offset, int count) => throw new InvalidOperationException(SR.IO_NotSupported_UnwritableStream);
+
+        public override void Flush() => throw new InvalidOperationException(SR.IO_NotSupported_UnwritableStream);
```

Comment:
This should be a nop.

Reply:
Thanks for letting me know. Will fix that.